packages/babel-cli/src/babel/dir.js

```diff
@@ -48,15 +47,10 @@ export default async function ({
     const dest = getDest(relative, base);
 
     try {
-      const res = await util.compile(
-        src,
-        defaults(
-          {
-            sourceFileName: slash(path.relative(dest + "/..", src)),
-          },
-          babelOptions,
-        ),
-      );
+      const res = await util.compile(src, {
```

Comment:
Shouldn't it be the other way around? Since `defaults`:

> Once a property is set, additional values of the same property are ignored.

Reply:
I'm finding it a little tricky to reason about the differences - in one way, yep, it seems like it may simply be a case of reversing the argument order.

As I understand it though, the spread operator will *always* use the keys discovered in later arguments to override the contents of previously-declared keys.

So `sourceFileName` from the default here would always end up overriding any values provided in `babelOptions`.  Are they always guaranteed to by the defaults and non-user-supplied?

Reply:
FWIW, it may not be the _intent_, but as far as I can tell, `sourceFileName` in this instance cannot be overridden by `babelOptions`?

Reply:
Yep, I think you're correct there, and I've updated the argument ordering here and the other callsite.

Because I'm still feeling a little uncertain about the change, I've marked it as a draft for now and will re-read the code again a bit later.